dracut/03flatcar-network/parse-ip-for-networkd.sh

```diff
@@ -1,17 +1,58 @@
-#!/bin/sh
+#!/bin/bash
 #
 # This script was mostly stolen from 40network/parse-ip-opts.sh.  Its
 # actions are adapted to write .network files to /etc/systemd/network
 # in the initramfs instead of using separate DHCP commands, etc.  Note
-# the bashisms.
+# the bashisms. It used to run as dracut hook, relying on afterburn to
+# process kargs for the dracut cmdline target but since that is too
+# early now that we start afterburn from /sysusr/ this script here
+# became an own systemd unit running shortly before systemd-networkd.
```

Comment:
```suggestion
# the bashisms. It used to be run as a dracut hook, relying on afterburn to
# process kargs for the dracut cmdline target. Now that we start afterburn
# from /sysusr instead of ramdisk root, we need to mount /sysusr first. Mounting
# happens later than the dracut hook we used. Thus we added a systemd unit for
# this script. The unit runs shortly before systemd-networkd.
```